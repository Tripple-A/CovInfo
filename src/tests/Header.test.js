import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Header from '../components/header';


const div = document.createElement('div');
const style = { display: 'block' };
const change = () => {
  if (style.display === 'block') {
    style.display = 'none';
  } else {
    style.display = 'block';
  }
};
const total = {
  updated: 1587123700636,
  cases: 2197161,
  todayCases: 15853,
  deaths: 147512,
  todayDeaths: 2041,
  recovered: 557617,
  active: 1492032,
  critical: 56517,
  casesPerOneMillion: 282,
  deathsPerOneMillion: 18,
  tests: 18507025,
  testsPerOneMillion: 2375.3,
  affectedCountries: 212,
};

it('renders without crashing', () => {
  render(
    <MemoryRouter>
      <Header total={total} style={style} change={change} />
    </MemoryRouter>, div,
  );
});

it('renders text correctly', () => {
  const { getByTestId } = render(
    <MemoryRouter>
      <Header total={total} style={style} change={change} />
    </MemoryRouter>, div,
  );
  expect(getByTestId('cov')).toHaveTextContent('Cov:');
});

it('matches snapshot', () => {
  const tree = renderer.create(
    <MemoryRouter>
      <Header total={total} style={style} change={change} />
    </MemoryRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
