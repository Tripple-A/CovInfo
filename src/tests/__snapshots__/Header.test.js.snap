// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="jumbotron"
>
  <ul
    className="nav"
  >
    <li
      className="nav-item logo"
      data-testid="cov"
    >
      Cov:
      <span
        className="info"
      >
        :Info 
      </span>
    </li>
    <li
      className="nav-item ml-auto mt-2"
    >
      Get the latest Covid-19 stats here
    </li>
  </ul>
  <div
    className="marquee"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Hide Stats
    </button>
    <p
      style={
        Object {
          "display": "block",
        }
      }
    >
      Total Cases:
      2197161
       
      , Total Deaths:
      147512
       
      , Total Recovered:
      557617
       
      , Total Tests:
      18507025
       
      , Total Active:
      1492032
    </p>
  </div>
</div>
`;
